lib/webmachine.mli

```diff
@@ -126,33 +136,128 @@ module type S = sig
+        [`OPTIONS] requests.
+
+        {i Default}: [\[("allow", self#allowed_methods)\]] *)
+
     method known_methods : (Code.meth list, 'body) op
+    (** A request to this resource whose method is not included in the returned
+        list will result in a [501 Not Implemented].
+
+        {i Default}: [\[`GET; `HEAD; `POST; `PUT; `DELETE; `Other "TRACE"; `Other "CONNECT"; `OPTIONS\]] *)
+
+    method allowed_methods : (Code.meth list, 'body) op
+    (** A request to this resource whose method is not included in the returned
+        list will result in a [405 Method Not Allowed]. The response will
+        include an ["allow"] header that lists the allowed methods.
+
+        {i Default}: [\[`GET; `HEAD\]] *)
+
     method delete_resource : (bool, 'body) op
+    (** [`DELETE] requests will call this method. Returning [true] indicates
+        that the deletion succeeded.
+
+        {i Default}: [false] *)
+
     method delete_completed : (bool, 'body) op
+    (** Only successful {{!delete_resource}delete_resource} calls will result
+        in a call to this method. Return [false] if the deletion was accepted
+        but cannot yet be guaranteed to have finished.
+
+        {i Default}: [true] *)
+
     method process_post : (bool, 'body) op
+    (** [`POST] requests will call this method. Returning true indicates the
+         POST succeeded. *)
+
     method language_available : (bool, 'body) op
+    (** Returning [false] will result in a [406 Not Acceptable].
+
+        {i Default}: [true] *)
+
     method charsets_provided : ((string * ('body -> 'body)) list, 'body) op
     method encodings_provided : ((string * ('body -> 'body)) list, 'body) op
+
     method variances : (string list, 'body) op
+    (** Returns a list of header names that should be included in the
+        response's [Vary] header. The standard content negotiation headers
+        (i.e., [Accept], [Accept-Encoding], [Accept-Charset], [Accept-Language])
+        will automatically be added if needed, and therefore do not need to be
+        specified here.
+
+        {i Default}: [[]] *)
+
     method is_conflict : (bool, 'body) op
+    (** Returning [true] will result in a [409 Conflict].
+
+        {i Default}: false *)
+
     method multiple_choices : (bool, 'body) op
+    (** Returning [true] indicates that multiple representations of the
+        resource exist and a single one cannot be automatically chosen, resulting
+        in a [300 Multiple Choices] rather than a [200 OK].
+
+        {i Default}: false *)
+
     method previously_existed : (bool, 'body) op
     method moved_permanently : (Uri.t option, 'body) op
     method moved_temporarily : (Uri.t option, 'body) op
+
     method last_modified : (string option, 'body) op
     method expires : (string option, 'body) op
+
     method generate_etag : (string option, 'body) op
+    (** Returning [Some string] will use [string] for the [ETag] header.
+
+        {i Default}: [None] *)
+
```

Comment:
I don't suppose there is support for the different types of ETags here? Maybe adding a type around string similar to this 
https://github.com/helium/airship/blob/master/src/Airship/Types.hs#L82-L90

Happy to have a go at that if you think it'll be useful.
@seliopou